mne/io/edf/edf.py

```diff
@@ -130,59 +130,68 @@ def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
-                            # don't resample tal_channels,
-                            # pad with zeros instead.
-                            n_missing = int(buf_len - n_samp)
-                            ch_data = np.hstack([ch_data, [0] * n_missing])
-                            ch_data = ch_data[r_sidx:r_eidx]
+                            # don't resample tal_channels, zero-pad instead.
+                            if n_samps[ci] < buf_len:
+                                z = np.zeros((len(ch_data),
+                                              buf_len - n_samps[ci]))
+                                ch_data = np.append(ch_data, z, -1)
+                            else:
+                                ch_data = ch_data[:, :buf_len]
                         elif ci == stim_channel:
                             if annot and annotmap or tal_channels is not None:
-                                # don't bother with resampling the stim ch
-                                # because it gets overwritten later on.
-                                ch_data = np.zeros(n_buf_samp)
+                                # don't resample, it gets overwritten later
+                                ch_data = np.zeros((len(ch_data, buf_len)))
                             else:
                                 # Stim channel will be interpolated
-                                oldrange = np.linspace(0, 1, n_samp + 1, True)
-                                newrange = np.linspace(0, 1, buf_len, False)
-                                newrange = newrange[r_sidx:r_eidx]
-                                ch_data = interp1d(
-                                    oldrange, np.append(ch_data, 0),
-                                    kind='zero')(newrange)
+                                old = np.linspace(0, 1, n_samps[ci] + 1, True)
+                                new = np.linspace(0, 1, buf_len, False)
+                                ch_data = np.append(
+                                    ch_data, np.zeros((len(ch_data), 1)), -1)
+                                ch_data = interp1d(old, ch_data,
+                                                   kind='zero', axis=-1)(new)
                         else:
-                            ch_data = resample(ch_data, buf_len, n_samp,
-                                               npad=0)[r_sidx:r_eidx]
-                    this_data[ii, :n_buf_samp] = ch_data
-                data[:, d_sidx:d_eidx] = this_data[:, :n_buf_samp]
+                            # XXX resampling each chunk isn't great,
+                            # it forces edge artifacts to appear at
+                            # each buffer boundary :(
+                            # it can also be very slow...
+                            ch_data = resample(
+                                ch_data, buf_len, n_samps[ci], npad=0, axis=-1)
+                    assert ch_data.shape == (len(ch_data), buf_len)
```

Comment:
This isn't a test, it's just a reminder / "active comment". We have this in a few other places already :)